File: Python/Kode/MusicInstrument.py
from Item import Item

class MusicInstrument(Item):
    def __init__(self, id=0, price=0.0, stock=0,
                 playing_method="", condition="", weight=""):
        # memanggil constructor Item
        super().__init__(id, price, stock)
        self.playing_method = playing_method
        self.condition = condition
        self.weight = weight

    # getter and setter playingMethod
    def set_playing_method(self, playing_method):
        self.playing_method = playing_method

    def get_playing_method(self):
        return self.playing_method

    # getter and setter condition
    def set_condition(self, condition):
        self.condition = condition

    def get_condition(self):
        return self.condition

    # getter and setter weight
    def set_weight(self, weight):
        self.weight = weight

    def get_weight(self):
        return self.weight

File: Java/Kode/MusicInstrument.java
public class MusicInstrument extends Item {
    private String playingMethod; // dipukul / ditabuh / dipetik
    private String condition; // likenew / baru / bekas / ori / premium
    private String weight; // weight disimpan sebagai String supaya ikut satuan "kg"

    // constructor kosong
    public MusicInstrument() {
        super(); // memanggil constructor kosong milik Item
        this.playingMethod = "";
        this.condition = "";
        this.weight = "";
    }

    // constructor berparameter (memanggil constructor Item lewat super())
    public MusicInstrument(int id, double price, int stock,
                            String playingMethod, String condition, String weight) {
        super(id, price, stock);
        this.playingMethod = playingMethod;
        this.condition = condition;
        this.weight = weight;
    }

    // getter and setter playingMethod
    public void setPlayingMethod(String playingMethod) { this.playingMethod = playingMethod; }
    public String getPlayingMethod() { return playingMethod; }

    // getter and setter condition
    public void setCondition(String condition) { this.condition = condition; }
    public String getCondition() { return condition; }

    // getter and setter weight
    public void setWeight(String weight) { this.weight = weight; }
    public String getWeight() { return weight; }
}